src/NuGet.Services.EndToEnd/Support/PushedPackagesFixture.cs

```diff
@@ -284,38 +286,39 @@ private async Task<List<PackageToPrepare>> InitializePackagesAsync(PackageType p
                                         GetPackageId(PackageType.SemVer2Prerel),
                                         VersionRange.Parse(SemVer2PrerelVersion))
                                 })
-                        }
+                        },
+                        Properties = new PackageProperties(PackageType.SemVer2DueToSemVer2Dep)
                     }));
                     break;
 
                 case PackageType.SemVer2StableMetadataUnlisted:
                     packageToPrepare = new PackageToPrepare(
-                        Package.Create(id, "1.0.0", "1.0.0+metadata"),
+                        Package.Create(id, "1.0.0", "1.0.0+metadata", PackageType.SemVer2StableMetadataUnlisted),
                         unlist: true);
                     break;
 
                 case PackageType.SemVer2StableMetadata:
-                    packageToPrepare = new PackageToPrepare(Package.Create(id, "1.0.0", "1.0.0+metadata"));
+                    packageToPrepare = new PackageToPrepare(Package.Create(id, "1.0.0", "1.0.0+metadata", PackageType.SemVer2StableMetadata));
                     break;
 
                 case PackageType.SemVer2PrerelUnlisted:
                     packageToPrepare = new PackageToPrepare(
-                        Package.Create(id, "1.0.0-alpha.1"),
+                        Package.Create(id, "1.0.0-alpha.1", PackageType.SemVer2PrerelUnlisted),
                         unlist: true);
                     break;
 
                 case PackageType.SemVer2Prerel:
-                    packageToPrepare = new PackageToPrepare(Package.Create(id, SemVer2PrerelVersion));
+                    packageToPrepare = new PackageToPrepare(Package.Create(id, SemVer2PrerelVersion, PackageType.SemVer2Prerel));
                     break;
 
                 case PackageType.SemVer2PrerelRelisted:
-                    packageToPrepare = new PackageToPrepare(Package.Create(id, "1.0.0-alpha.1"),
-                        unlist: true);
+                    packageToPrepare = new PackageToPrepare(Package.Create(id, "1.0.0-alpha.1", PackageType.SemVer2PrerelRelisted),
+                        unlist: true); 
                     break;
 
                 case PackageType.SemVer1StableUnlisted:
                     packageToPrepare = new PackageToPrepare(
-                        Package.Create(id, "1.0.0"),
+                        Package.Create(id, "1.0.0", PackageType.SemVer1StableUnlisted),
```

Comment:
nit; in all the changes above to the `PackageToPrepare` simply pass the `packageType` which is the same as the ones you are qualifying explicitly.

Reply:
The same issue as above.